TEMP_CCAAS_TUTORIAL.md

```diff
@@ -0,0 +1,137 @@
+
+Chaincode-as-service requires you to orchestrate the build and deployment phase yourself. Whilst this is an additional step, it gives control back. The Peer still requires a 'chaincode package' to be installed. In this case this doesn't contain code, but the information about where the chaincode is hosted. (Hostname,Port,TLS config etc)
+
+## Fabric v2.4.1 Improvements
+
+We need to use the latest 2.4.1 release as this contains some improvements to make this process easier. The core functionality is available in earlier releases but requires more configuration.
+
+- The docker image for the peer contains a builder for chaincode-as-a-service preconfigured. This is named 'ccaasbuilder'. This removes the need to build your own external builder and repackage and configure the peer
+- The `ccaasbuilder` applications are included in the binary tgz archive download for use in other circumstances. The `sampleconfig/core.yaml` is updated as well to refer to 'ccaasbuilder'
+- The 2.4.1 Java Chaincode release has been updated to remove the need to write a custom bootstrap main class. (similar  to the NodeJS Chaincode. It is intended that this will be added to the go chaincode as well.
+
+## End-to-end with the the test-network
+
+The `test-network` and some of the chaincodes have been updated to support running chaincode-as-a-service. The commands below assume that you've got the latest fabric-samples cloned, along with the latest Fabric docker images. 
+
+It's useful to have two terminal windows open, one for starting the Fabric Network, and a second for monitoring all the docker containers.
+
+In your 'monitoring' window, run this to watch all activity from the all the docker containers on the `fabric_test` network
+
+```bash
+# from the fabric-samples repo
+./test-network/monitordocker.sh
+```
+In the 'Fabric Network' window, start the test network
+
+```bash
+cd test-network
+./network.sh up createChannel
+```
+
+You can run other variants of this command, eg to use CouchDB or CAs, without affecting the '-as-a-service' feature. The three keys steps are:
+
+- Build a docker image of the contract. Both `/asset-transfer-basic/chaincode-typescript` and `/asset-transfer-basic/chaincode-java` have been updated with Dockerfiles
+- Install, Approve and Commit a chaincode definition. This is unchanged, but the chaincode package contains connection inforamtion, not code
+- Start the docker container(s) containing the contract
+
+Note that the order listed isn't mandatory. The key thing is that the containers are runing before the first transaction is set by the peer.
+
+This sequence can be run as follows
+```
+./network.sh deployCCAAS  -ccn basicts -ccp ../asset-transfer-basic/chaincode-typescript 
+```
```

Comment:
This step failed for me. When building docker image I get this error:

```
Step 5/20 : RUN npm ci && npm run package
 ---> Running in 549576af86e6
npm ERR! The `npm ci` command can only install with an existing package-lock.json or
npm ERR! npm-shrinkwrap.json with lockfileVersion >= 1. Run an install with npm@5 or
npm ERR! later to generate a package-lock.json file, then try again.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-12-15T02_45_29_696Z-debug.log
The command '/bin/sh -c npm ci && npm run package' returned a non-zero code: 1
Docker buid of chaincode-as-a-service container failed
Deploying chaincode-as-a-service failed
```

Reply:
I think the shrinkwrap file wasn't copied in.. it is now. (I hope)